src/main/scala/view/scalaFX/FXControllers/ChartChoiceController.scala

```diff
@@ -14,19 +15,30 @@ sealed trait ChartChoiceControllerInterface {
 }
 
 @sfxml
-class ChartChoiceController( private val pedigreeRadioButton: RadioButton ) extends ChartChoiceControllerInterface {
+class ChartChoiceController( private val pedigreeRadioButton: RadioButton,
+                             private val legendBox: VBox) extends ChartChoiceControllerInterface {
 
   private var baseAppController: Option[BaseAppControllerInterface] = None
 
-  override def initialize(controller: BaseAppControllerInterface): Unit = baseAppController = Some(controller)
+  override def initialize(controller: BaseAppControllerInterface): Unit = {
+    legendBox.setVisible(false)
+    baseAppController = Some(controller)
+  }
 
   override def handleBunnyClick(): Unit = if (pedigreeRadioButton.selected.value) baseAppController --> { _.showPedigreeChart() }
 
-  def showPopulationChart(): Unit = baseAppController --> { _.showPopulationChart() }
-
-  def showMutationsChart(): Unit = baseAppController --> { _.showProportionsChart() }
-
-  def showPedigreeChart(): Unit = baseAppController --> { _.showPedigreeChart() }
+  def showPopulationChart(): Unit = {
+    legendBox.setVisible(false)
+    baseAppController --> { _.showPopulationChart() }
+  }
 
+  def showMutationsChart(): Unit = {
+    legendBox.setVisible(false)
+    baseAppController --> { _.showProportionsChart() }
+  }
 
+  def showPedigreeChart(): Unit = {
+    legendBox.setVisible(true)
+    baseAppController --> { _.showPedigreeChart() }
+  }
```

Comment:
Rifattorizzare usando un metodo "generico" ti può piacere? 
```suggestion
  def showChart(legendVisibility: Boolean, chartToShow : BaseAppController => Unit) {
    legendBox.setVisible(legendVisibility)
    baseAppController --> { chartToShow }
  }
```

Reply:
Non so se ho capito bene la domanda, però ti direi di sì è da richiamare dentro i metodi per mostrare i chart, esempio: showPopulationsChart